### ISSUE TITLE
[java] OnlyOneReturn: False positive with equals method

### ISSUE THREAD

**Issue by sarser2048** [pmd/pmd#4257]
**Affects PMD Version:**
6.51.0+
7.0.0

**Rule:** [OnlyOneReturn](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#onlyonereturn)

**Description:**
When used PMD, I found many false positives about the rule OnlyOneReturn. I think we usually have multi return statements in `equals` method.

For example, the annotation `EqualsAndHashCode` from lombok usually generate many return statements in `equals` method. There are also many examples in large java projects, e.g., [Spring](https://github.com/spring-projects/spring-boot/blob/b17eb9d6b599976a1a7d21987a0d7175ac1a1f8f/spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/filewatch/FileSnapshot.java#L52).

**Code Sample demonstrating the issue:**
```java
@Override
public boolean equals(Object o) {  // a case generated by the lombok annotation
    if (o == this) return true;
    if (!(o instanceof EqualsAndHashCodeExample)) return false;
    EqualsAndHashCodeExample other = (EqualsAndHashCodeExample) o;
    if (!other.canEqual((Object)this)) return false;
    if (this.getName() == null ? other.getName() != null : !this.getName().equals(other.getName())) return false;
    if (Double.compare(this.score, other.score) != 0) return false;
    if (!Arrays.deepEquals(this.tags, other.tags)) return false;
    return true;
  }
```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.

**Running PMD through:** *[Maven]*

**Comment by adangel** [comment]
We could indeed add a property to ignore `equals` by default.
The similar checkstyle rule [ReturnCount](https://checkstyle.org/config_coding.html#ReturnCount) ignores equals by default (see property format).
